torch/_dynamo/variables/torch.py

```diff
@@ -1149,6 +1208,34 @@ def speculate_branch(branch):
             example_value = deepcopy_to_fake_tensor(example_res, tx.fake_mode)
 
             p_args = (lowered_node,) + p_args
+        elif self.value.__name__ == "wrap":
+            # See NOTE [HigherOrderOperator tracing design] for more details
+            checkpoint = tx.copy_graphstate()
+            graph_checkpoint = tx.output.graph
+            (
+                body_r,
+                body_graph,
+                body_lifted_freevars,
+            ) = speculate_subgraph(
+                args[0],
+                [
+                    *args[1:],
+                ],
+                graph_checkpoint,
+                checkpoint,
+            )
+
+            body_name = add_subgraph(
+                "body", torch.fx.GraphModule(tx.output.nn_modules, body_graph)
+            )
```

Comment:
(probably dumb) nit: it looks like `add_subgraph` hardcodes the name of the submodule to be `cond_{name}_{i}`. Is `add_subgraph` general infra meant for use by all higher order ops?